src/transformers/models/t5/modeling_t5.py

```diff
@@ -689,6 +690,12 @@ def forward(
 
         outputs = (hidden_states,)
 
+        if present_key_value_state is None and self.training:
```

Comment:
I think it would be nicer if we can slightly change the code to **not** even return `present_key_value_state` if `use_cache=False`. (You could check how it's done for GPT2, e.g. :-))

Reply:
Yes, it's indeed a cleaner approach. I've updated the code.

Reply:
Thanks for updating!